projects/extension/src/containers/Options.tsx

```diff
@@ -57,6 +60,7 @@ const Options: React.FunctionComponent = () => {
 
   useEffect(() => {
     if (!bg) return
+    setClientError(bg?.uiInterface.smoldotCrashError)
```

Comment:
This should happen after you call `onSmoldotCrashErrorChanged`, otherwise, if there's an `await` somewhere in between, the crash might happen during this `await`.

Reply:
Maybe I was not clear before. There are 2 cases that we need to cover in order to show the error in the extension's UI.
The first case is when the UI is open and the error occurs (which is covered [by the onSmoldotCrashErrorChanged](https://github.com/paritytech/substrate-connect/pull/1228/files#diff-a091fcf55d424bd3b3ee40fced674890335fb4753e0fe0f88722e7b986cdac43R121-R123)), and when the error occurs while the Popup or Options are not open. 
The `setClientError` in the code here (or the way it was prior to the previous change), runs as soon as the page inits, and sets the outcome of the `smoldotCrashError` to the `clientError` state variable. If this code is not there, then there may be a crashed Client prior to opening any extension page/popup, but the user will never realize as the error will not appear

Reply:
I understand all what you're saying. This is not related to my comment.
If `bg.uiInterface.smoldotCrashError` is `undefined` when you call `setClientError`, then becomes defined, and then only you call `onSmoldotCrashErrorChanged`, you won't detect that the value has been defined.